de.dlr.sc.virsat.model.extension.tests.test/src/de/dlr/sc/virsat/model/extension/tests/model/json/TestCategoryReferenceTest.java

```diff
@@ -0,0 +1,161 @@
+		refPropBean.getATypeInstance().setUuid(new VirSatUuid("0dee3e78-fbcd-4294-8dba-5fa3d4760249"));
+		
+		refCatBean = tcReference.getTestRefCategoryBean();
+		refCatBean.getATypeInstance().setUuid(new VirSatUuid("f34d30b0-80f5-4c96-864f-29ab4d3ae9f2"));
+		tcAllProperty = new TestCategoryAllProperty(concept);
+		JsonTestHelper.setTestCategoryAllPropertyUuids(tcAllProperty);
+		JsonTestHelper.createRepositoryWithUnitManagement(concept);
+		
+		bpString.setValue(JsonTestHelper.TEST_STRING);
+		tcReference.setTestRefCategory(tcAllProperty);
+	}
+	
+	/**
+	 * Marshalls the test object and asserts that the result equals the test resource
+	 * @param resource the test resource
+	 * @throws JAXBException
+	 * @throws IOException
+	 */
+	private void assertMarshall(String resource) throws JAXBException, IOException {
+		Marshaller jsonMarshaller = jaxbUtility.getJsonMarshaller();
+		
+		StringWriter sw = new StringWriter();
+		jsonMarshaller.marshal(tcReference, sw);
+		
+		String expectedJson = TestActivator.getResourceContentAsString(resource);
+		assertEqualsNoWs("Json is as expected", expectedJson, sw.toString());
+	}
+	
+	@Test
+	public void testJsonMarshalling() throws JAXBException, IOException {
+		
+		assertMarshall(RESOURCE);
+	}
+	
+	@Test
+	public void testJsonUnmarshalling() throws JAXBException, IOException {
+		BeanPropertyString bpString2 = new TestCategoryAllProperty(concept).getTestStringBean();
+		bpString2.getATypeInstance().setUuid(new VirSatUuid("1256e7a2-9a1f-443c-85f8-7b766eac3f50"));
+		
+		// Quick mock setup to embed the model into a resource set
+		ResourceSet resourceSet = new ResourceSetImpl();
+		Resource resourceImpl = new ResourceImpl();
+		resourceSet.getResources().add(resourceImpl);
+		resourceImpl.getContents().add(tcReference.getATypeInstance());
+		resourceImpl.getContents().add(bpString.getATypeInstance());
+		resourceImpl.getContents().add(bpString2.getATypeInstance());
+		resourceImpl.getContents().add(refCatBean.getATypeInstance());
+		resourceImpl.getContents().add(tcAllProperty.getATypeInstance());
+		
+		Unmarshaller jsonUnmarshaller = jaxbUtility.getJsonUnmarshaller(resourceSet);
+		
+		String inputJson = TestActivator.getResourceContentAsString(RESOURCE_CHANGED_REFERENCE);
+		StringReader sr = new StringReader(inputJson);
+
+		assertEquals(bpString.getUuid(), tcReference.getTestRefProperty().getUuid());
+		
+		jsonUnmarshaller.unmarshal(new StreamSource(sr), TestCategoryReference.class);
+		
+		assertEquals("Referenced bean changed successfully", bpString2.getUuid(), tcReference.getTestRefProperty().getUuid());
+	}
+	
+	@Test
+	public void testJsonMarshallingNull() throws JAXBException, IOException {
+		
+		tcReference.setTestRefProperty(null);
+		tcReference.setTestRefCategory(null);
+		
+		assertMarshall(RESOURCE_NULL_REFERENCE);
+	}
+	
+	@Test
+	public void testJsonUnmarshallingNull() throws JAXBException, IOException {
+		
+		// Quick mock setup to embed the model into a resource set
```

Comment:
A lot of repetitive code in the unmarshalling tests, maybe we can extract some of it into a helper method or something?

Reply:
I added two methods to the JsonTestHelper, to reduce repetitive code